configuration/pih/liquibase/sql/create_functions.sql

```diff
@@ -2127,3 +2129,54 @@ RETURN ret;
 
 END
 #
+
+-- The following function accepts encounter_id
+-- It will return the names of the creator of the encounter (names of the data entry clerk)
+-- as saved in the encounter table
+#
+DROP FUNCTION IF EXISTS encounter_creator;
+#
+CREATE FUNCTION encounter_creator(
+    _encounter_id int
+)
+    RETURNS TEXT
+    DETERMINISTIC
+
+BEGIN
+    DECLARE creatorName TEXT;
+
+    SELECT CONCAT(given_name, ' ', family_name) creator_names INTO creatorName
+    FROM person_name pn
+    JOIN users u ON pn.voided = 0 AND u.retired = 0 AND pn.person_id = u.person_id
+    JOIN encounter e ON e.creator = u.user_id AND e.voided = 0 AND e.encounter_id = _encounter_id;
+
+    RETURN creatorName;
```

Comment:
Each user likely has only one name, but what happens if for some reason a user has two names?

Reply:
thanks @mogoodrich , I added in a preffered = 1 condition